csrc/host_ir/executor.cpp

```diff
@@ -281,6 +281,25 @@ void HostIrExecutor::handle(EndCoalescing* end_coalescing) {
   works_[end_coalescing] = backend->endCoalescing();
 }
 
+void HostIrExecutor::handle(MatmulOp* matmul) {
+  TensorView* a = matmul->inA();
+  TensorView* b = matmul->inB();
+  TensorView* out = matmul->out();
+  NVF_ERROR(
+      expr_evaluator_.isKnown(a) && expr_evaluator_.isKnown(b),
+      "Inputs of the matmul ",
+      matmul->toString(),
+      "must be precomputed before being retrieved");
+  if (expr_evaluator_.isKnown(out)) {
```

Comment:
Thanks -- this is part of #3181 that I'd like to separate out. 

Can you try to address https://github.com/NVIDIA/Fuser/pull/3181#discussion_r1800222396? I love small unit tests with good coverage -- much easier to debug than larger ones.

Reply:
We already cover both branches with two separate unit tests. 

- The pre-existing `TEST_F(MatmulHostIrTest, HostIr)` https://github.com/NVIDIA/Fuser/blob/55971f8811243822983edf8510b3fa26524d1f15/tests/cpp/test_host_irs.cpp#L759  tests when we do not pre-allocate the output
- the newly added unit tests with pre-allocated outputs

Reply:
Great!